include/Keymap.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Keymap.h - holds information about a key mapping
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef KEYMAP_H
+#define KEYMAP_H
+
+#include <vector>
+#include <QObject>
+#include <QString>
+
+#include "Note.h"
+#include "SerializingObject.h"
+
+class Keymap : public QObject, public SerializingObject
+{
+	Q_OBJECT
+public:
+	Keymap();
+	Keymap(	QString description,
+			std::vector<int> newMap,
+			int newFirst,
+			int newLast,
+			int newMiddle,
+			int newBaseKey,
+			float newBaseFreq);
+
+	QString getDescription() const;
+	void setDescription(QString description);
+
+	int getMiddleKey() const {return m_middleKey;}
+	int getFirstKey() const {return m_firstKey;}
+	int getLastKey() const {return m_lastKey;}
+	int getBaseKey() const {return m_baseKey;}
+	float getBaseFreq() const {return m_baseFreq;}
+
+	int getSize() const {return m_map.size();}
+	int getDegree(int key) const;
+	int getOctave(int key) const;
+	const std::vector<int> &getMap() const {return m_map;}
```

Comment:
I always try to use the _standard_ library if I cannot find a clear and good reason why the Qt version should be used instead. I really hate how Qt obsessively tries to reinvent the wheel for every single thing, so whenever possible I keep using the standard features of C++, because I still consider LMMS to be written in C++, not Qt++. I mean: where do you stop? Should we, for example, start using `qreal` instead of `float` for everything?

Reply:
I agree to avoid Qt in the core because of [realtime-safety](https://github.com/LMMS/lmms/wiki/Realtime-conventions#Qt-Data-structures).